tests/dataset_builders/pie/tbga/test_tbga.py

```diff
@@ -0,0 +1,264 @@
+            "the top 1% down-regulated genes in PCa.",
+            "relation": "NA",
+            "h": {"id": 50940, "name": "PDE11A", "pos": [82, 6]},
+            "t": {"id": "C0006826", "name": "Malignant Neoplasms", "pos": [26, 6]},
+        }
+    else:
+        raise ValueError(f"Invalid split: {split}")
+
+
+def test_example_to_document(hf_example, split):
+    doc = example_to_document(hf_example)
+    if split == "test":
+        assert doc.entities.resolve() == [
+            ("50940", "PDE11A", "PDE11A"),
+            ("C0006826", "Malignant Neoplasms", "cancer"),
+        ]
+        assert doc.relations.resolve() == [
+            ("NA", (("50940", "PDE11A", "PDE11A"), ("C0006826", "Malignant Neoplasms", "cancer")))
+        ]
+
+    elif split == "train":
+        assert doc.entities.resolve() == [
+            ("6347", "CCL2", "monocyte chemoattractant protein-1"),
+            ("C0231221", "Asymptomatic", "asymptomatic"),
+        ]
+        assert doc.relations.resolve() == [
+            (
+                "NA",
+                (
+                    ("6347", "CCL2", "monocyte chemoattractant protein-1"),
+                    ("C0231221", "Asymptomatic", "asymptomatic"),
+                ),
+            )
+        ]
+
+    elif split == "validation":
+        assert doc.entities.resolve() == [
+            ("51726", "DNAJB11", "ERdj3"),
+            ("C0000768", "Congenital Abnormality", "malformation"),
+        ]
+        assert doc.relations.resolve() == [
+            (
+                "NA",
+                (
+                    ("51726", "DNAJB11", "ERdj3"),
+                    ("C0000768", "Congenital Abnormality", "malformation"),
+                ),
+            )
+        ]
+
+    else:
+        raise ValueError(f"Unknown split variant: {split}")
+
+
+@pytest.mark.slow
+def test_example_to_document_all(hf_dataset, split):
+    for example in hf_dataset[split]:
+        doc = example_to_document(example)
+        assert doc is not None
+
+
+@pytest.fixture(scope="module")
+def builder() -> BUILDER_CLASS:
+    return BUILDER_CLASS()
+
+
+@pytest.fixture(scope="module")
+def generated_document(builder, hf_example):
+    return builder._generate_document(hf_example)
+
+
+def test_builder(builder):
+    assert builder is not None
+    assert builder.dataset_name == DATASET_NAME
+    assert builder.document_type == TbgaDocument
+
+
+def test_document_to_example(generated_document, builder, hf_example):
+    hf_example_back = builder._generate_example(generated_document)
+    assert hf_example_back == hf_example
+
+
+@pytest.fixture(scope="module")
+def pie_dataset(split):
+    ds = load_pie_dataset(str(PIE_DATASET_PATH), split=split)
+    return ds
+
+
+@pytest.mark.slow
+def test_pie_dataset(pie_dataset, split):
+    assert pie_dataset is not None
+    assert len(pie_dataset) == SPLIT_SIZES[split]
+    for doc in pie_dataset:
+        # cannot assert real document type "BioRelDocument" (look also test_imdb.py)
```

Comment:
```suggestion
        # cannot assert real document type "TbgaDocument" because it comes from dataset script copied to the huggingface cache folder (and not from its origin in dataset_builders/pie/tbga)
```